docs/DeveloperGuide.adoc

```diff
@@ -0,0 +1,665 @@
+
+=== Edit Task Feature
+==== Implementation
+The `EditCommand` class extends the `Command` class by providing functions to edit specific tasks in the list of
+*ATAS*.
+
+Given below is an example usage scenario of the `edit` command.
+
+*Step 1* +
+The user types in `edit 1`. The `parseCommand()` method of the `Parser` class is called to obtain `edit` which is the type
+of command the user is entering.
+
+[WARNING]
+An `IncorrectCommand` class will be returned and an `UNKNOWN_COMMAND_ERROR` string from the `Messages` class will be passed
+into the constructor of that class if the command supplied was invalid.
+
+*Step 2* +
+The `parseCommand()` method subsequently calls the `prepareEditCommand()` method inside the same `Parser` class. This method
+splits the `fullCommand` string parameters into 2 tokens. The integer `1` will be obtained as the *Index* of the task
+specified in the list. This method returns a new instance of the `EditCommand` class, passing the integer `1` as the parameter.
+
+[WARNING]
+An `IncorrectCommand` class will be returned and a `NUM_FORMAT_ERROR` string from the `Messages` class will be passed
+into the constructor of that class if the number supplied was not an *integer*. +
+An `IncorrectCommand` class will be returned and a `INCORRECT_ARGUMENT_ERROR` string from the `Messages` class will be passed
+into the constructor of that class if there are no task index supplied by the user. +
+
+*Step 3* +
+A new instance of the `EditCommand` class is returned to the main method of *ATAS* with parameter `1` as described above.
+The `execute()` method of the `EditCommand` class is now called.
+
+*Step 4* +
+The `execute()` method in the `EditCommand` class first gets an input from the user on the details of the edited task.
+
+[TIP]
+Assignment Command Format: `assignment n/[NAME] m/[MODULE] d/DD/MM/YY HHmm c/[COMMENTS]` +
+Event Command Format: `event n/[NAME] l/[LOCATION] d/DD/MM/YY HHmm - HHmm c/[COMMENTS]`
+
+*Step 5* +
+If the user supplies an `assignment` command, the `editAssignment()` method will be invoked. This method extracts the
+`assignmentName`, `moduleName`, `dateTime` and `comments` string to return a new instance of the `Assignment` class. +
+
+If the user supplies an `event` command, the `editEvent()` method will be invoked. This method extracts the
+`eventName`, `location`, `startDateTime`, `endDateTime` and `comments` string to return a new instance of the `Event` class.
+
+*Step 6* +
+This newly instanced class (either `Assignment` or `Event`) will be passed into the method `editTask()` of the `TaskList` class.
+The `editTask()` method of the `TaskList` class uses Java's `ArrayList` `set()` method to replace the task.
+
+*Step 7* +
+Finally, a `CommandResult` class is returned with `EDIT_SUCCESS_MESSAGE` passed as the parameter to the constructor of
+that class.
+
+The following sequence diagram summarises what happens when the `EditCommand` class is executed.
+
+image::EditCommand_SequenceDiagram.png[]
```

Comment:
Shouldn't the self invocation arrows point to the start of the activation bar, instead of a point below it?

![Annotation 2020-03-31 124310](https://user-images.githubusercontent.com/59698024/77987804-5ee58e80-734d-11ea-9a9b-7957f34f1f64.jpg)